# Pandas 3 Ödev

In [1]:
import pandas as pd
import numpy as np


1) df'in içindeki sütunların birbirleriyle olan korelasyonlarına(ilişkilerine) bakın.


In [3]:
df= pd.read_csv("mpg.csv")
df.head()

mpg  cylinders  displacement horsepower  weight  acceleration  model_year  \
0  18.0          8         307.0        130    3504          12.0          70   
1  15.0          8         350.0        165    3693          11.5          70   
2  18.0          8         318.0        150    3436          11.0          70   
3  16.0          8         304.0        150    3433          12.0          70   
4  17.0          8         302.0        140    3449          10.5          70   

   origin                       name  
0       1  chevrolet chevelle malibu  
1       1          buick skylark 320  
2       1         plymouth satellite  
3       1              amc rebel sst  
4       1                ford torino

In [6]:
korelasyon = df.corr()
korelasyon

mpg  cylinders  displacement    weight  acceleration  \
mpg           1.000000  -0.775396     -0.804203 -0.831741      0.420289   
cylinders    -0.775396   1.000000      0.950721  0.896017     -0.505419   
displacement -0.804203   0.950721      1.000000  0.932824     -0.543684   
weight       -0.831741   0.896017      0.932824  1.000000     -0.417457   
acceleration  0.420289  -0.505419     -0.543684 -0.417457      1.000000   
model_year    0.579267  -0.348746     -0.370164 -0.306564      0.288137   
origin        0.563450  -0.562543     -0.609409 -0.581024      0.205873   

              model_year    origin  
mpg             0.579267  0.563450  
cylinders      -0.348746 -0.562543  
displacement   -0.370164 -0.609409  
weight         -0.306564 -0.581024  
acceleration    0.288137  0.205873  
model_year      1.000000  0.180662  
origin          0.180662  1.000000

2) mpg sütunu ile diğer sütunların arasındaki korelasyonları özel olarak inceleyin.(Bunu göstermedik ama df.corr() çağırdıktan sonra aynı normal df'lerde olduğu gibi istediğiniz sütunu çağırabiliyorsunuz.)


In [13]:
korelasyon["mpg"]

mpg             1.000000
cylinders      -0.775396
displacement   -0.804203
weight         -0.831741
acceleration    0.420289
model_year      0.579267
origin          0.563450
Name: mpg, dtype: float64

3) cylinders sütununda hangi unique değerler olduğuna liste halinde bakın.

In [15]:
uni_cyl = df["cylinders"].unique()
uni_cyl

array([8, 4, 6, 3, 5], dtype=int64)

4) cylinders sütunundaki unique değerlerin her birinden kaçar adet olduğuna bakın.

In [19]:
df["cylinders"].value_counts()


4    204
8    103
6     84
3      4
5      3
Name: cylinders, dtype: int64

5) Arabanın modeli 75-80 yılları arasındaki arabaların hepsini getirin.(75 ve 80 dahil)

In [20]:
df[df["model_year"].between(75,80)]

mpg  cylinders  displacement horsepower  weight  acceleration  \
152  19.0          6         225.0         95    3264          16.0   
153  18.0          6         250.0        105    3459          16.0   
154  15.0          6         250.0         72    3432          21.0   
155  15.0          6         250.0         72    3158          19.5   
156  16.0          8         400.0        170    4668          11.5   
..    ...        ...           ...        ...     ...           ...   
333  32.7          6         168.0        132    2910          11.4   
334  23.7          3          70.0        100    2420          12.5   
335  35.0          4         122.0         88    2500          15.1   
336  23.6          4         140.0          ?    2905          14.3   
337  32.4          4         107.0         72    2290          17.0   

     model_year  origin                     name  
152          75       1  plymouth valiant custom  
153          75       1           chevrolet nova  
154          75       1          mercury monarch  
155          75       1            ford maverick  
156          75       1         pontiac catalina  
..          ...     ...                      ...  
333          80       3            datsun 280-zx  
334          80       3            mazda rx-7 gs  
335          80       2        triumph tr7 coupe  
336          80       1       ford mustang cobra  
337          80       3             honda accord  

[186 rows x 9 columns]

6) horsepower'ı 155-180 arasındaki arabaların hepsini getirin.(155 dahil, 180 dahil değil)(Bu sorguyu yapabilmeniz için horsepower sütununu object durumdan float duruma getirmeniz gerekiyor.)


In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           398 non-null    float64
 1   cylinders     398 non-null    int64  
 2   displacement  398 non-null    float64
 3   horsepower    398 non-null    object 
 4   weight        398 non-null    int64  
 5   acceleration  398 non-null    float64
 6   model_year    398 non-null    int64  
 7   origin        398 non-null    int64  
 8   name          398 non-null    object 
dtypes: float64(3), int64(4), object(2)
memory usage: 28.1+ KB


In [22]:
df["horsepower"].astype(float)

ValueError: could not convert string to float: '?'

In [23]:
df[df["horsepower"] == '?']

mpg  cylinders  displacement horsepower  weight  acceleration  \
32   25.0          4          98.0          ?    2046          19.0   
126  21.0          6         200.0          ?    2875          17.0   
330  40.9          4          85.0          ?    1835          17.3   
336  23.6          4         140.0          ?    2905          14.3   
354  34.5          4         100.0          ?    2320          15.8   
374  23.0          4         151.0          ?    3035          20.5   

     model_year  origin                  name  
32           71       1            ford pinto  
126          74       1         ford maverick  
330          80       2  renault lecar deluxe  
336          80       1    ford mustang cobra  
354          81       2           renault 18i  
374          82       1        amc concord dl

In [28]:
df["horsepower"].replace("?", np.nan, inplace= True)

In [29]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           398 non-null    float64
 1   cylinders     398 non-null    int64  
 2   displacement  398 non-null    float64
 3   horsepower    392 non-null    object 
 4   weight        398 non-null    int64  
 5   acceleration  398 non-null    float64
 6   model_year    398 non-null    int64  
 7   origin        398 non-null    int64  
 8   name          398 non-null    object 
dtypes: float64(3), int64(4), object(2)
memory usage: 28.1+ KB


In [34]:
df["horsepower"] = df["horsepower"].astype(float)

In [35]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           398 non-null    float64
 1   cylinders     398 non-null    int64  
 2   displacement  398 non-null    float64
 3   horsepower    392 non-null    float64
 4   weight        398 non-null    int64  
 5   acceleration  398 non-null    float64
 6   model_year    398 non-null    int64  
 7   origin        398 non-null    int64  
 8   name          398 non-null    object 
dtypes: float64(4), int64(4), object(1)
memory usage: 28.1+ KB


7) Elinizdeki df'i horsepower'a göre büyükten küçüğe sıralayın ardından df'in indexlerini sıfırlayın.

In [39]:
sort_df = df.sort_values(by="horsepower", ascending= False)
sort_df

mpg  cylinders  displacement  horsepower  weight  acceleration  \
116  16.0          8         400.0       230.0    4278           9.5   
95   12.0          8         455.0       225.0    4951          11.0   
13   14.0          8         455.0       225.0    3086          10.0   
8    14.0          8         455.0       225.0    4425          10.0   
6    14.0          8         454.0       220.0    4354           9.0   
..    ...        ...           ...         ...     ...           ...   
126  21.0          6         200.0         NaN    2875          17.0   
330  40.9          4          85.0         NaN    1835          17.3   
336  23.6          4         140.0         NaN    2905          14.3   
354  34.5          4         100.0         NaN    2320          15.8   
374  23.0          4         151.0         NaN    3035          20.5   

     model_year  origin                      name  
116          73       1        pontiac grand prix  
95           73       1  buick electra 225 custom  
13           70       1   buick estate wagon (sw)  
8            70       1          pontiac catalina  
6            70       1          chevrolet impala  
..          ...     ...                       ...  
126          74       1             ford maverick  
330          80       2      renault lecar deluxe  
336          80       1        ford mustang cobra  
354          81       2               renault 18i  
374          82       1            amc concord dl  

[398 rows x 9 columns]

In [42]:
sort_df.reset_index(inplace=True, drop=True)
sort_df

mpg  cylinders  displacement  horsepower  weight  acceleration  \
0    16.0          8         400.0       230.0    4278           9.5   
1    12.0          8         455.0       225.0    4951          11.0   
2    14.0          8         455.0       225.0    3086          10.0   
3    14.0          8         455.0       225.0    4425          10.0   
4    14.0          8         454.0       220.0    4354           9.0   
..    ...        ...           ...         ...     ...           ...   
393  21.0          6         200.0         NaN    2875          17.0   
394  40.9          4          85.0         NaN    1835          17.3   
395  23.6          4         140.0         NaN    2905          14.3   
396  34.5          4         100.0         NaN    2320          15.8   
397  23.0          4         151.0         NaN    3035          20.5   

     model_year  origin                      name  
0            73       1        pontiac grand prix  
1            73       1  buick electra 225 custom  
2            70       1   buick estate wagon (sw)  
3            70       1          pontiac catalina  
4            70       1          chevrolet impala  
..          ...     ...                       ...  
393          74       1             ford maverick  
394          80       2      renault lecar deluxe  
395          80       1        ford mustang cobra  
396          81       2               renault 18i  
397          82       1            amc concord dl  

[398 rows x 9 columns]

8) Elinizdeki df'den en küçük 5 acceralation'a sahip olan arabaları getirin.

In [44]:
df.nsmallest(5, columns="acceleration")

mpg  cylinders  displacement  horsepower  weight  acceleration  \
11  14.0          8         340.0       160.0    3609           8.0   
7   14.0          8         440.0       215.0    4312           8.5   
9   15.0          8         390.0       190.0    3850           8.5   
6   14.0          8         454.0       220.0    4354           9.0   
12  15.0          8         400.0       150.0    3761           9.5   

    model_year  origin                   name  
11          70       1     plymouth 'cuda 340  
7           70       1      plymouth fury iii  
9           70       1     amc ambassador dpl  
6           70       1       chevrolet impala  
12          70       1  chevrolet monte carlo

---

In [55]:
df1 = pd.read_csv("CIA_Country_Facts.csv")
df1.head()

Country                               Region  Population  \
0     Afghanistan        ASIA (EX. NEAR EAST)             31056997   
1         Albania  EASTERN EUROPE                          3581655   
2         Algeria  NORTHERN AFRICA                        32930091   
3  American Samoa  OCEANIA                                   57794   
4         Andorra  WESTERN EUROPE                            71201   

   Area (sq. mi.)  Pop. Density (per sq. mi.)  Coastline (coast/area ratio)  \
0          647500                        48.0                          0.00   
1           28748                       124.6                          1.26   
2         2381740                        13.8                          0.04   
3             199                       290.4                         58.29   
4             468                       152.1                          0.00   

   Net migration  Infant mortality (per 1000 births)  GDP ($ per capita)  \
0          23.06                              163.07               700.0   
1          -4.93                               21.52              4500.0   
2          -0.39                               31.00              6000.0   
3         -20.71                                9.27              8000.0   
4           6.60                                4.05             19000.0   

   Literacy (%)  Phones (per 1000)  Arable (%)  Crops (%)  Other (%)  Climate  \
0          36.0                3.2       12.13       0.22      87.65      1.0   
1          86.5               71.2       21.09       4.42      74.49      3.0   
2          70.0               78.1        3.22       0.25      96.53      1.0   
3          97.0              259.5       10.00      15.00      75.00      2.0   
4         100.0              497.2        2.22       0.00      97.78      3.0   

   Birthrate  Deathrate  Agriculture  Industry  Service  
0      46.60      20.34        0.380     0.240    0.380  
1      15.11       5.22        0.232     0.188    0.579  
2      17.14       4.61        0.101     0.600    0.298  
3      22.46       3.27          NaN       NaN      NaN  
4       8.71       6.25          NaN       NaN      NaN

9) Elimizdeki CIA'in ülkeler hakkında topladığı basit açık kaynak verileri df'inde kaç ülkenin verisi olduğunu bulun.


In [56]:
df1.shape

(227, 20)

10) Elimizdeki sütunlardan hangilerinde kaçar tane nan veri olduğunu bulun.


In [57]:
nan_var = df1.isna()
nan_var

Country  Region  Population  Area (sq. mi.)  Pop. Density (per sq. mi.)  \
0      False   False       False           False                       False   
1      False   False       False           False                       False   
2      False   False       False           False                       False   
3      False   False       False           False                       False   
4      False   False       False           False                       False   
..       ...     ...         ...             ...                         ...   
222    False   False       False           False                       False   
223    False   False       False           False                       False   
224    False   False       False           False                       False   
225    False   False       False           False                       False   
226    False   False       False           False                       False   

     Coastline (coast/area ratio)  Net migration  \
0                           False          False   
1                           False          False   
2                           False          False   
3                           False          False   
4                           False          False   
..                            ...            ...   
222                         False          False   
223                         False           True   
224                         False          False   
225                         False          False   
226                         False          False   

     Infant mortality (per 1000 births)  GDP ($ per capita)  Literacy (%)  \
0                                 False               False         False   
1                                 False               False         False   
2                                 False               False         False   
3                                 False               False         False   
4                                 False               False         False   
..                                  ...                 ...           ...   
222                               False               False          True   
223                                True                True          True   
224                               False               False         False   
225                               False               False         False   
226                               False               False         False   

     Phones (per 1000)  Arable (%)  Crops (%)  Other (%)  Climate  Birthrate  \
0                False       False      False      False    False      False   
1                False       False      False      False    False      False   
2                False       False      False      False    False      False   
3                False       False      False      False    False      False   
4                False       False      False      False    False      False   
..                 ...         ...        ...        ...      ...        ...   
222              False       False      False      False    False      False   
223               True       False      False      False    False       True   
224              False       False      False      False    False      False   
225              False       False      False      False    False      False   
226              False       False      False      False    False      False   

     Deathrate  Agriculture  Industry  Service  
0        False        False     False    False  
1        False        False     False    False  
2        False        False     False    False  
3        False         True      True     True  
4        False         True      True     True  
..         ...          ...       ...      ...  
222      False        False     False    False  
223       True         True      True    False  
224      False        False     False    False  
225      False        False     False    Fa

In [58]:
nan_var.sum()

Country                                0
Region                                 0
Population                             0
Area (sq. mi.)                         0
Pop. Density (per sq. mi.)             0
Coastline (coast/area ratio)           0
Net migration                          3
Infant mortality (per 1000 births)     3
GDP ($ per capita)                     1
Literacy (%)                          18
Phones (per 1000)                      4
Arable (%)                             2
Crops (%)                              2
Other (%)                              2
Climate                               22
Birthrate                              3
Deathrate                              4
Agriculture                           15
Industry                              16
Service                               15
dtype: int64

11) Elimizde tamamen nan değerlerden oluşan bir satır varsa o satırı düşürün.(Tamamen nan değerlerden oluşan bir satır olup olmadığını, herhangi bir satırın düşüp düşmediğini kontrol edin)


In [59]:
df1.dropna(how="all")

Country                               Region  Population  \
0       Afghanistan        ASIA (EX. NEAR EAST)             31056997   
1           Albania  EASTERN EUROPE                          3581655   
2           Algeria  NORTHERN AFRICA                        32930091   
3    American Samoa  OCEANIA                                   57794   
4           Andorra  WESTERN EUROPE                            71201   
..              ...                                  ...         ...   
222       West Bank  NEAR EAST                               2460492   
223  Western Sahara  NORTHERN AFRICA                          273008   
224           Yemen  NEAR EAST                              21456188   
225          Zambia  SUB-SAHARAN AFRICA                     11502010   
226        Zimbabwe  SUB-SAHARAN AFRICA                     12236805   

     Area (sq. mi.)  Pop. Density (per sq. mi.)  Coastline (coast/area ratio)  \
0            647500                        48.0                          0.00   
1             28748                       124.6                          1.26   
2           2381740                        13.8                          0.04   
3               199                       290.4                         58.29   
4               468                       152.1                          0.00   
..              ...                         ...                           ...   
222            5860                       419.9                          0.00   
223          266000                         1.0                          0.42   
224          527970                        40.6                          0.36   
225          752614                        15.3                          0.00   
226          390580                        31.3                          0.00   

     Net migration  Infant mortality (per 1000 births)  GDP ($ per capita)  \
0            23.06                              163.07               700.0   
1            -4.93                               21.52              4500.0   
2            -0.39                               31.00              6000.0   
3           -20.71                                9.27              8000.0   
4             6.60                                4.05             19000.0   
..             ...                                 ...                 ...   
222           2.98                               19.62               800.0   
223            NaN                                 NaN                 NaN   
224           0.00                               61.50               800.0   
225           0.00                               88.29               800.0   
226           0.00                               67.69              1900.0   

     Literacy (%)  Phones (per 1000)  Arable (%)  Crops (%)  Other (%)  \
0            36.0                3.2       12.13       0.22      87.65   
1            86.5               71.2       21.09       4.42      74.49   
2            70.0               78.1        3.22       0.25      96.53   
3            97.0              259.5       10.00      15.00      75.00   
4           100.0              497.2        2.22       0.00      97.78   
..            ...                ...         ...        ...        ...   
222           NaN              145.2       16.90      18.97      64.13   
223           NaN                NaN        0.02       0.00      99.98   
224          50.2               37.2        2.78       0.24      96.98   
225          80.6                8.2        7.08       0.03      92.90   
226          90.7               26.8        8.32       0.34      91.34   

     Climate  Birthrate  Deathrate  Agriculture  Industry  Service  
0        1.0      46.60      20.34        0.380     0.240    0.380  
1        3.0      15.11       5.22        0.232     0.188    0.579  
2        1.0      17.14       4.61        0.101     0.600    0.298  
3        2.0      22.46       3.27          NaN       NaN      NaN  
4  

Koddan önce çalıştırılan Shape metoduna göre (227,20) shape'inde verş bulunurken, dropna() metodundan sonra da shape aynı kaldı. Bu durum dataframe de tüm değerleri Nan olan bir satır yok.

12) Her Bölgedeki(Region) toplam kişi sayısına bakın.


In [63]:
Reg_Pop = df1[["Region","Population"]]
Reg_Pop

Region  Population
0          ASIA (EX. NEAR EAST)             31056997
1    EASTERN EUROPE                          3581655
2    NORTHERN AFRICA                        32930091
3    OCEANIA                                   57794
4    WESTERN EUROPE                            71201
..                                   ...         ...
222  NEAR EAST                               2460492
223  NORTHERN AFRICA                          273008
224  NEAR EAST                              21456188
225  SUB-SAHARAN AFRICA                     11502010
226  SUB-SAHARAN AFRICA                     12236805

[227 rows x 2 columns]

In [64]:
Reg_Pop.groupby("Region").sum()

Population
Region                                         
ASIA (EX. NEAR EAST)                 3687982236
BALTICS                                 7184974
C.W. OF IND. STATES                   280081548
EASTERN EUROPE                        119914717
LATIN AMER. & CARIB                   561824599
NEAR EAST                             195068377
NORTHERN AFRICA                       161407133
NORTHERN AMERICA                      331672307
OCEANIA                                33131662
SUB-SAHARAN AFRICA                    749437000
WESTERN EUROPE                        396339998

13) Climate değerleri nan olanlar, tüm df'in ortalama climate değeriyle doldurun.(Gerçek bir analiz yapıyor olsaydık bu çok mantıklı bir hareket olmazdı, belki her bulunduğu bölgenin ortalama climate değeriyle doldurmak daha mantıklı bir hareket olurdu. Fakat şimdilik bu kadar kasmaya gerek yok. nan değerleri tüm climate ortalamalarıyla doldursanız yeterli.)


In [71]:
nan_deger_say = df1["Climate"].isna().value_counts()[1]
nan_deger_say

22

**CHALLENGE: Gerçek bir analiz yapıyor olsaydık bu çok mantıklı bir hareket olmazdı, belki her bulunduğu bölgenin ortalama climate değeriyle doldurmak daha mantıklı bir hareket olurdu.**

In [67]:
Reg_Cli = df1[["Region","Climate"]]
Reg_Cli

Region  Climate
0          ASIA (EX. NEAR EAST)               1.0
1    EASTERN EUROPE                           3.0
2    NORTHERN AFRICA                          1.0
3    OCEANIA                                  2.0
4    WESTERN EUROPE                           3.0
..                                   ...      ...
222  NEAR EAST                                3.0
223  NORTHERN AFRICA                          1.0
224  NEAR EAST                                1.0
225  SUB-SAHARAN AFRICA                       2.0
226  SUB-SAHARAN AFRICA                       2.0

[227 rows x 2 columns]

In [84]:
cli_mean = Reg_Cli.groupby("Region").mean()
cli_mean

Climate
Region                                       
ASIA (EX. NEAR EAST)                 1.962963
BALTICS                              3.000000
C.W. OF IND. STATES                  2.550000
EASTERN EUROPE                       3.111111
LATIN AMER. & CARIB                  2.033333
NEAR EAST                            1.666667
NORTHERN AFRICA                      1.500000
NORTHERN AMERICA                     2.000000
OCEANIA                              2.000000
SUB-SAHARAN AFRICA                   1.885417
WESTERN EUROPE                       3.095238

In [90]:
Reg_Cli = Reg_Cli.fillna(Reg_Cli.groupby("Region").transform('mean'))


In [91]:
Reg_Cli["Climate"].isna().value_counts()

False    227
Name: Climate, dtype: int64

In [93]:
Reg_Cli[Reg_Cli["Climate"] == 2.550000]

Region  Climate
137  C.W. OF IND. STATES      2.55
169  C.W. OF IND. STATES      2.55

14) Bölgeler ve Ülkelere göre gruplama yapın ardından Doğum miktarlarına bakın.


In [102]:
birthrate = df1.groupby(["Region", "Country"])[["Birthrate"]].mean()
birthrate

Birthrate
Region                              Country                  
ASIA (EX. NEAR EAST)                Afghanistan         46.60
                                    Bangladesh          29.80
                                    Bhutan              33.65
                                    Brunei              18.79
                                    Burma               17.91
...                                                       ...
WESTERN EUROPE                      San Marino          10.02
                                    Spain               10.06
                                    Sweden              10.27
                                    Switzerland          9.71
                                    United Kingdom      10.71

[227 rows x 1 columns]

15) Veriyi istediğiniz şekilde inceleyerek bu zamana kadar öğrendiğiniz bilgilerle anlamlı 5 adet yargı cümlesi çıkartın. ( En çok insan Asya kıtasında yaşıyor, En çok insan Afrinın sahralarında ölüyor, Ortalama en yüksek telefon sahibi Kuzey Amerikada, En çok okuma yazma oranına sahip ülkeler Baltık ülkeleri vb.)


In [108]:
birthrate.nsmallest(1, "Birthrate")

Birthrate
Region                        Country             
ASIA (EX. NEAR EAST)          Hong Kong       7.29

In [109]:
birthrate.nlargest(1, "Birthrate")

Birthrate
Region                              Country           
SUB-SAHARAN AFRICA                  Niger        50.73

In [110]:
Reg_Pop.nlargest(1, "Population")

Region  Population
42  ASIA (EX. NEAR EAST)           1313973713

In [111]:
Reg_Pop.nsmallest(1, "Population")

Region  Population
174  NORTHERN AMERICA                           7026

In [125]:
df1.groupby("Region")[["Country"]].count()

Country
Region                                      
ASIA (EX. NEAR EAST)                      28
BALTICS                                    3
C.W. OF IND. STATES                       12
EASTERN EUROPE                            12
LATIN AMER. & CARIB                       45
NEAR EAST                                 16
NORTHERN AFRICA                            6
NORTHERN AMERICA                           5
OCEANIA                                   21
SUB-SAHARAN AFRICA                        51
WESTERN EUROPE                            28

In [149]:
sorted_df = df1.sort_values(by="GDP ($ per capita)", ascending=False)
result_df = sorted_df[(sorted_df["Population"] < 10000)] 
result_df

Country                               Region  Population  \
174  St Pierre & Miquelon  NORTHERN AMERICA                           7026   
140            Montserrat              LATIN AMER. & CARIB            9439   
171          Saint Helena  SUB-SAHARAN AFRICA                         7502   

     Area (sq. mi.)  Pop. Density (per sq. mi.)  Coastline (coast/area ratio)  \
174             242                        29.0                         49.59   
140             102                        92.5                         39.22   
171             413                        18.2                         14.53   

     Net migration  Infant mortality (per 1000 births)  GDP ($ per capita)  \
174          -4.86                                7.54              6900.0   
140           0.00                                7.35              3400.0   
171           0.00                               19.00              2500.0   

     Literacy (%)  Phones (per 1000)  Arable (%)  Crops (%)  Other (%)  \
174          99.0              683.2       13.04        0.0      86.96   
140          97.0                NaN       20.00        0.0      80.00   
171          97.0              293.3       12.90        0.0      87.10   

     Climate  Birthrate  Deathrate  Agriculture  Industry  Service  
174      NaN      13.52       6.83          NaN       NaN      NaN  
140      2.0      17.59       7.10          NaN       NaN      NaN  
171      NaN      12.13       6.53          NaN       NaN      NaN

Doğum oranı en düşük olan ülke Hong Kong -- 7.29

Doğum oranı en yüksek olan ülke Niger -- 50.73

En kalabalık Bölge: ASIA (EX. NEAR EAST), nüfus: 1,313,973,713

Nüfusu en az olan Bölge: NORTHERN AMERICA, nüfus: 7,026

Nüfusu 10,000 kişinin altında olup GDP ($ per capita) değeri en yüksek  olan ülke 

    St Pierre & Miquelon (Pop: 7026,  GDP ($ per capita): 6,900.0)